core/src/main/java/io/opencensus/tags/TagContextFactory.java

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright 2017, Google Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opencensus.tags;
+
+/**
+ * Factory for new {@link TagContext}s and {@code TagContext}s based on the current context.
+ */
+public abstract class TagContextFactory {
```

Comment:
What about TagContextBuilder?

Also where are you going to have a method like:
Scope withTagContext(TagContext)?

Does it make sense to have a Tagger as well (similar with Tracer)?